samples/snippets/src/test/java/pubsublite/spark/SamplesIntegrationTest.java

```diff
@@ -0,0 +1,309 @@
+import com.google.cloud.pubsublite.SubscriptionName;
+import com.google.cloud.pubsublite.SubscriptionPath;
+import com.google.cloud.pubsublite.TopicName;
+import com.google.cloud.pubsublite.TopicPath;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.google.common.flogger.GoogleLogger;
+import com.google.pubsub.v1.PubsubMessage;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Queue;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.junit.Before;
+import org.junit.Test;
+
+public class SamplesIntegrationTest extends SampleTestBase {
+
+  private static final GoogleLogger log = GoogleLogger.forEnclosingClass();
+
+  private SubscriptionName sourceSubscriptionName;
+  private SubscriptionPath sourceSubscriptionPath;
+  private TopicName destinationTopicId;
+  private TopicPath destinationTopicPath;
+  private SubscriptionName destinationSubscriptionName;
+  private SubscriptionPath destinationSubscriptionPath;
+  private Boolean initialized = false;
+
+  @Before
+  public void beforeClass() throws Exception {
+    if (initialized) {
+      return;
+    }
+    log.atInfo().log("RunId is: %s", runId);
+    setupEnvVars();
+    findMavenHome();
+
+    // Maven package into jars
+    mavenPackage(workingDir);
+    mavenPackage(workingDir + "/samples");
+
+    // Upload to GCS
+    Storage storage =
+        StorageOptions.newBuilder().setProjectId(projectId.value()).build().getService();
+    uploadGCS(storage, sampleJarNameInGCS, sampleJarLoc);
+    uploadGCS(storage, connectorJarNameInGCS, connectorJarLoc);
+    initialized = true;
+  }
+
+  /** Note that source single word messages have been published to a permanent topic. */
+  @Test
+  public void testWordCount() throws Exception {
+    UUID testId = UUID.randomUUID();
+    setupSourceWithTestId(testId);
+    setupDestinationWithTestId(testId);
+    try {
+      // Run Dataproc job, block until it finishes
+      SparkJob.Builder sparkJobBuilder =
+          SparkJob.newBuilder()
+              .setMainClass("pubsublite.spark.WordCount")
+              .putProperties("spark.submit.deployMode", "cluster")
+              .putProperties(
+                  "spark.yarn.appMasterEnv.SOURCE_SUBSCRIPTION_PATH",
+                  sourceSubscriptionPath.toString())
+              .putProperties(
+                  "spark.yarn.appMasterEnv.DESTINATION_TOPIC_PATH",
+                  destinationTopicPath.toString());
+      runDataprocJob(sparkJobBuilder);
+      // Verify final destination messages in Pub/Sub Lite
+      verifyWordCountResultViaPSL();
+    } finally {
+      deleteSubscriptionExample(cloudRegion.value(), sourceSubscriptionPath);
+      deleteSubscriptionExample(cloudRegion.value(), destinationSubscriptionPath);
+      deleteTopicExample(cloudRegion.value(), destinationTopicPath);
+    }
```

Comment:
Is the intention here to include these as part of the IT instead of thinking them as part of the teardown (`@After`)?

Reply:
`@After` is the same that applies to all tests, while in our case, each tests needs to create different sets of topic/subs and clean them up slightly differently.